applications: serial_lte_modem: improve HTTPC service (#4178)
nrfconnect/sdk-nrf: applications/serial_lte_modem/doc/HTTPC_AT_commands.rst

  Each header field should end with ``<CR><LF>``.
* The ``<payload_length>`` is an integer.
  It represents the length of the payload.

>   If ``payload_length`` is greater than ``0``, the SLM will enter the data mode and expect the upcoming UART input data as payload.

Reviewer: ```suggestion
  If ``payload_length`` is greater than ``0``, the SLM will enter data mode and expect the upcoming UART input data as payload.
```

  The SLM will then send the payload to the HTTP server until the ``payload_length`` bytes are sent.

>   To abort sending the payload, use ``CONFIG_SLM_DATAMODE_TERMINATOR`` to terminate data mode. The default pattern string is "+++".
>   UART silence ``CONFIG_SLM_DATAMODE_SILENCE`` before and after the pattern string is used to exit data mode.

Reviewer: ```suggestion
  To abort sending the payload, terminate data mode by sending the terminator string defined in :option:`CONFIG_SLM_DATAMODE_TERMINATOR`.
  The default pattern string is "+++".
  Keep in mind that UART silence as configured in :option:`CONFIG_SLM_DATAMODE_SILENCE` is required before and after the pattern string.
```

Author: The description was changed per suggestion.

  It represents the length of a partially received HTTP response.
* The ``<state>`` value can assume one of the following values:

  * ``0`` - There is more HTTP response data to come.
  * ``1`` - The entire HTTP response has been received.

* The ``<response>`` is the raw data of the HTTP response, including headers and body.

Example
~~~~~~~

> The following example sends a PUT request to send JSON format data to the server with an optional header.

Reviewer: ```suggestion
The following example sends a PUT request to send data in JSON format to the server, with an optional header.
```

Author: The description was changed per suggestion.

::

>    at#xhttpccon=1,"iot.cht.com.tw",80

Reviewer: ```suggestion
   AT#XHTTPCCON=1,"iot.cht.com.tw",80
```

Reviewer: Can we use example.com as URL?

Author: Yes, example.com  would be more suitable. I've modified it, thanks.
